components/database.py

```diff
@@ -511,10 +609,12 @@ def get_number_of_students_taking_module_in_ay_sem(module_code, ay_sem):
                   "WHERE sp.moduleCode = %s " + \
                   "AND sp.acadYearAndSem = %s " + \
                   "ORDER BY COUNT(*) DESC"
-
-    DB_CURSOR.execute(sql_command, (module_code, ay_sem))
-
-    return DB_CURSOR.fetchone()[0]
+    try:
+        DB_CURSOR.execute(sql_command, (module_code, ay_sem))
+        return DB_CURSOR.fetchone()[0]
+    except psycopg2.Error:
+        CONNECTION.rollback()
+        return []
```

Comment:
As the try block returns a number, I think the except clause should return a number too ?